docs/startup/preview/index.md

```diff
@@ -0,0 +1,529 @@
+                                                     pageView -> pageView.getRegion())
+pageView.sendTo(userRegions);
+{% endhighlight %}
+
+
+### Sink
+Allows sending messages from this MessageStream to an output system using the provided [SinkFunction](/learn/documentation/{{site.version}}/api/javadocs/org/apache/samza/operators/functions/SinkFunction.html).
+
+This offers more control than sendTo since the SinkFunction has access to the `MessageCollector` and the `TaskCoordinator`. For instance, you can choose to manually commit offsets, or shut-down the job using the TaskCoordinator APIs. This operator can also be used to send messages to non-Samza systems (e.g. remote databases or REST services etc.)
+
+{% highlight java %}
+// Repartition pageView by userId
+MessageStream<PageView> pageViews = ...
+pageViews.sink( (msg, collector, coordinator) -> {
+// Construct a new outgoing message, and send it to a kafka topic named TransformedPageViewEvent
+ collector.send(new OutgoingMessageEnvelope(new SystemStream(“kafka”,
+                         “TransformedPageViewEvent”), msg));
+} )
+{% endhighlight %}
+
+### Join
+
+The Join operator joins messages from two MessageStreams using the provided pairwise [JoinFunction](/learn/documentation/{{site.version}}/api/javadocs/org/apache/samza/operators/functions/JoinFunction.html). Messages are joined when the keys extracted from messages from the first stream match keys extracted from messages in the second stream. Messages in each stream are retained for the provided ttl duration and join results are emitted as matches are found.
+
+{% highlight java %}
+// Joins a stream of OrderRecord with a stream of ShipmentRecord by orderId with a TTL of 20 minutes.
+// Results are produced to a new stream of FulfilledOrderRecord.
+MessageStream<OrderRecord> orders = …
+MessageStream<ShipmentRecord> shipments = …
+
+
+MessageStream<FulfilledOrderRecord> shippedOrders = orders.join(shipments, new OrderShipmentJoiner(), Duration.ofMinutes(20) )
+
+// Constructs a new FulfilledOrderRecord by extracting the order timestamp from the OrderRecord and the shipment timestamp from the ShipmentRecord.
+
+ class OrderShipmentJoiner implements JoinFunction<String, OrderRecord, ShipmentRecord, FulFilledOrderRecord> {
+   @Override
+   public FulFilledOrderRecord apply(OrderRecord message, ShipmentRecord otherMessage) {
+     return new FulFilledOrderRecord(message.orderId, message.orderTimestamp, otherMessage.shipTimestamp);
+   }
+
+   @Override
+   public String getFirstKey(OrderRecord message) {
+     return message.orderId;
+   }
+
+   @Override
+   public String getSecondKey(ShipmentRecord message) {
+     return message.orderId;
+   }
+ }
+
+{% endhighlight %}
+
+
+### Window
+
+#### Windowing Concepts
+**Windows, Triggers and WindowPanes**: The window operator groups incoming messages in the MessageStream into finite windows. Each emitted result contains one or more messages in the window and is called a WindowPane.
```

Comment:
> Windows, Triggers and WindowPane

Unless Triggers and WindowPane are a single thing, you might want to stick an oxford comma here:

> Windows, Triggers, and WindowPane